pinot-core/src/test/java/com/linkedin/pinot/core/io/util/PinotDataBitSetTest.java

```diff
@@ -27,6 +27,25 @@
   private static final Random RANDOM = new Random();
   private static final int NUM_ITERATIONS = 1000;
 
+  @Test
+  public void testGetNumBitsPerValue() {
+    for (int i = 0; i < NUM_ITERATIONS; i++) {
+      int cardinality = RANDOM.nextInt(Integer.MAX_VALUE) + 1;
+      int actual = PinotDataBitSet.getNumBitsPerValue(cardinality);
+      if (cardinality == 1) {
```

Comment:
Since this is an edge case, can we check separately when cardinality = 1? Currently, it will hit 1 with very very low chance.

Reply:
Good point. Added